js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1158,36 +1158,46 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
             return true;
         }
 
-        if(!value && this._isLastRequired(node)) {
+        if(value === false && this._isLastRequired(node)) {
             if(this._showCheckboxes()) {
                 const $node = this._getNodeElement(node);
-                const checkbox = this._getCheckBoxInstance($node);
-
-                checkbox && checkbox.option('value', true);
+                this._setCheckboxValue($node, true);
             }
             return false;
         }
 
-        const selectedNodesKeys = this.getSelectedNodeKeys();
-        if(this._isSingleSelection() && value) {
-            each(selectedNodesKeys, (index, nodeKey) => {
-                this.unselectItem(nodeKey);
+        if(value === true && this._isSingleSelection()) {
+            const selectedKeys = this.getSelectedNodeKeys();
+            each(selectedKeys, (index, key) => {
```

Comment:
There are only one key possible in single selection mode.
Refer to this [test](https://github.com/DevExpress/DevExtreme/pull/16797/files#diff-93b90b0dccd7cf8f9cc76c2579db3ffb418b310975d50d8cfdfc21f7fc88e240R1031)